crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -2740,6 +2750,35 @@ impl ClosureReplacer<'_> {
 }
 
 impl VisitMut for ClosureReplacer<'_> {
+    fn visit_mut_stmt(&mut self, stmt: &mut Stmt) {
+        stmt.visit_mut_children_with(self);
+
+        // Replace super calls in hoisted static class methods with empty
+        // statements so that we can remove them later to ensure that the
+        // resulting JS is still valid. Before this, we have already emitted an
+        // error for the super call.
+        if let Stmt::Expr(ExprStmt {
+            expr:
+                box Expr::Call(CallExpr {
+                    callee: Callee::Expr(box Expr::SuperProp(SuperPropExpr { .. })),
+                    ..
+                }),
+            ..
+        }) = stmt
```

Comment:
not sure why this is only targeting expression statements that contain a call, a `super.foo()` call can be arbitrarily deep within another expression

```
class A {
  static foo() { return 'A.foo' }
}

class B extends A {
  static foo() { return [super.foo(), 'and ', 'B.foo'].join(' ') }
}

B.foo() // "A.foo and B.foo"
```